scvi/external/contrastivevi/_model.py

```diff
@@ -0,0 +1,770 @@
+"""Model class for contrastive-VI for single cell expression data."""
+
+import logging
+import warnings
+from functools import partial
+from typing import Dict, Iterable, List, Optional, Sequence, Union
+
+import numpy as np
+import pandas as pd
+import torch
+from anndata import AnnData
+
+from scvi import REGISTRY_KEYS, settings
+from scvi.data import AnnDataManager
+from scvi.data.fields import (
+    CategoricalJointObsField,
+    CategoricalObsField,
+    LayerField,
+    NumericalJointObsField,
+    NumericalObsField,
+)
+from scvi.dataloaders import AnnDataLoader
+from scvi.model._utils import (
+    _get_batch_code_from_category,
+    _init_library_size,
+    scrna_raw_counts_properties,
+)
+from scvi.model.base import BaseModelClass
+from scvi.model.base._training_mixin import ContrastiveTrainingMixin
+from scvi.model.base._utils import _de_core
+from scvi.utils import setup_anndata_dsp
+
+from ._module import ContrastiveVAE
+
+logger = logging.getLogger(__name__)
+Number = Union[int, float]
+
+
+class ContrastiveVI(ContrastiveTrainingMixin, BaseModelClass):
+    """contrastive variational inference :cite:p:`Weinberger23`.
+
+    Parameters
+    ----------
+        adata
+            AnnData object that has been registered via
+            :meth:`~scvi.model.ContrastiveVI.setup_anndata`.
+        n_hidden
+            Number of nodes per hidden layer.
+        n_background_latent
+            Dimensionality of the background shared latent space.
+        n_salient_latent
+            Dimensionality of the salient latent space.
+        n_layers
+            Number of hidden layers used for encoder and decoder NNs.
+        dropout_rate
+            Dropout rate for neural networks.
+        use_observed_lib_size
+            Use observed library size for RNA as scaling factor in mean of conditional
+            distribution.
+        wasserstein_penalty
+            Weight of the Wasserstein distance loss that further discourages background
+            shared variations from leaking into the salient latent space.
```

Comment:
Needs to be unindented

Reply:
Good catch. I also unindented some other mistakenly indented docstrings.